etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/TransformationRuleEngine.java

```diff
@@ -32,26 +32,28 @@ private TransformationRuleEngine() {}
     @Override
     public void unloadRules() {
         rules.clear();
+        rulesLoaded = false;
     }
 
     @Override
     public void ensureRulesLoaded() throws RuleLoaderException {
-        // Double-checked locking - needed to protect from excessive sync locks
-        if (rules.isEmpty()) {
-            synchronized (this) {
+        if (!rulesLoaded) {
+            synchronized (rules) {
+                rulesLoaded = true;
```

Comment:
I just realized, the rules aren't loaded until `rules.addAll` is called.  So, unless I'm misunderstanding something, `rulesLoaded = true;` should be moved to right below the `rules.addAll` line.  Then we can remove the `rulesLoaded = false;` line that is right above the `throw new RuleLoaderException` line.

Reply:
Updated the code so the flag is where it should be.